indigo_app/views/documents.py

```diff
@@ -62,3 +62,6 @@ def get_context_data(self, **kwargs):
         context['download_formats'].sort(key=lambda f: f['title'])
 
         return context
+
+class DocumentPopupView(DocumentDetailView, DetailView):
```

Comment:
This is going to load DocumentDetailView's data unnecessarily.

Instead, make this

```python
class DocumentPopupView(AbstractAuthedIndigoView, DetailView):
    model = Document
    context_object_name = 'document'
    pk_url_kwarg = 'doc_id'
    template_name = 'indigo_api/document_popup.html'

    def get_object(self, queryset=None):
        doc = super(DocumentDetailView, self).get_object(queryset)
        if doc.deleted:
            raise Http404()
        return doc
```

and then copy in the

Reply:
Thanks!